### PR TITLE
feat: implement `eth_call` rpc endpoint

### LINKED ISSUE #176: Implement `eth_call` RPC endpoint

### PR DESCRIPTION
**Pull request by ElFantasma**
**Motivation**

Implement eth_call rpc endpoint

**Description**

Closes #176

### CODE REVIEW

File: crates/rpc/eth/transaction.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        Some(CallRequest {
+            transaction: serde_json::from_value(params[0].clone()).ok()?,
+            block: serde_json::from_value(params[1].clone()).ok()?,
+        })
+    }
+}
+
+pub fn call(request: &CallRequest, storage: Store) -> Result<Value, RpcErr> {
+    let block = request.block.clone().unwrap_or_default();
+    info!("Requested access list creation for tx on block: {}", block);
```
Comment (fmoletta) [inline crates/rpc/eth/transaction.rs:29]: Please update this message

Comment (ElFantasma) [reply crates/rpc/eth/transaction.rs:29]: oops, I saw it in logs and was sure I corrected it (more than once 🤔  )

**Review by avilagaston9 — APPROVED** [review]
LGTM!

File: crates/evm/evm.rs [hunk 2]
```diff
@@ -79,6 +83,31 @@ pub fn execute_tx(
...
+    header: &BlockHeader,
+    state: &mut EvmState,
+    spec_id: SpecId,
+) -> Result<ExecutionResult, EvmError> {
+    let block_env = block_env(header);
+    let tx_env = tx_env_from_generic(tx, header.base_fee_per_gas);
+    run_without_commit(tx_env, block_env, state, spec_id)
+}
+
+fn adjust_base_fee(
```
Comment (mpaulucci) [inline crates/evm/evm.rs:98]: where does this come from? A comment or link to a discussion on this could be useful.

File: crates/evm/evm.rs [hunk 3]
```diff
@@ -412,11 +455,19 @@ fn tx_env(tx: &Transaction) -> TxEnv {
 }
 
 // Used to estimate gas and create access lists
-fn tx_env_from_generic(tx: &GenericTransaction) -> TxEnv {
+fn tx_env_from_generic(tx: &GenericTransaction, basefee: u64) -> TxEnv {
+    let gas_price = if tx.gas_price != 0 {
```
Comment (mpaulucci) [inline crates/evm/evm.rs:459]: can we extract this to a function?

File: crates/rpc/eth/transaction.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            return Err(RpcErr::Revert {
+                data: format!("0x{:#x}", output),
+            });
+        }
+        ExecutionResult::Halt {
+            reason: _,
+            gas_used: _,
+        } => Bytes::new(),
+    };
+    serde_json::to_value(format!("0x{:#x}", data)).map_err(|_| RpcErr::Internal)
```
Comment (mpaulucci) [inline crates/rpc/eth/transaction.rs:68]: I think you can use "?" here instead of map_err

**Review by mpaulucci — APPROVED** [review]

File: crates/rpc/eth/transaction.rs (hunk 4 above)
Comment (ElFantasma) [reply crates/rpc/eth/transaction.rs:68]: But it expects a Result, using `?` will make it return Value, instead

**Pull request merged**